tests/atomistics/structure/test_atoms.py

```diff
@@ -667,8 +667,23 @@ def test_occupy_lattice(self):
         self.assertEqual(orig_basis.get_chemical_formula(), "Cl31HNa32")
 
     def test_select_index(self):
-        self.assertTrue(np.array_equal(self.CO2.select_index("C"), [0]))
-        self.assertTrue(np.array_equal(self.CO2.select_index("O"), [1, 2]))
+        basis = Atoms(symbols=['Fe', 'Cu', 'Ni', 'Al'], positions=np.random.random((4, 3)), cell=np.eye(3))
+        self.assertTrue(np.array_equal(basis.select_index("Fe"), [0]))
+        self.assertTrue(np.array_equal(basis.select_index("Ni"), [2]))
+        self.assertTrue(np.array_equal(basis.select_index(['Cu', 'Al']), [1, 3]))
+        Fe = basis.convert_element('Fe')
+        Ni = basis.convert_element('Ni')
+        self.assertTrue(np.array_equal(basis.select_index([Fe, Ni]), [0, 2]))
+        pse = PeriodicTable()
+        pse.add_element("Ni", "Ni_up", spin="up")
```

Comment:
Setting the spin as a string is possible but confusing, as in practise we commonly expect a float value or a vector.